gooresmed/upload.py

```diff
@@ -417,14 +431,21 @@ def _process_response(self, response):
             # Tombstone the current upload so it cannot be used again.
             self._finished = True
         elif status_code == PERMANENT_REDIRECT:
-            bytes_range = _helpers.header_required(response, u'range')
+            try:
+                bytes_range = _helpers.header_required(response, u'range')
+            except exceptions.InvalidResponse:
```

Comment:
I don't quite like this part, but felt it was the "lesser of evil". **Every** other usage of `header_required` just raises (on failure) so I didn't think it'd be worth using a return value for raising.